pkg/apis/application/v1alpha1/types.go

```diff
@@ -166,14 +165,45 @@ func (h *ApplicationSourceHelm) IsZero() bool {
 	return h == nil || (h.ReleaseName == "") && len(h.ValueFiles) == 0 && len(h.Parameters) == 0
 }
 
+type KustomizeImage string
+
+func (i KustomizeImage) delim() string {
+	if strings.Contains(string(i), "=") {
+		return "="
+	} else {
+		return ":"
+	}
+}
+
+// if the image name matches (i.e. up to the first "=" symbol)
+func (i KustomizeImage) Match(j KustomizeImage) bool {
```

Comment:
The logic looks much simpler than: https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/commands/edit/set/setimage.go#L158

Your logic looks correct to me, but maybe check to see what they are doing to make sure we aren't missing something.

Reply:
I think there is a third case `<image>=<new-image><:|@><new-tag>` (using `@`). This makes me think this is complex.